perma_web/perma/celery_tasks.py

```diff
@@ -1224,3 +1228,74 @@ def conditionally_queue_internet_archive_uploads_for_date_range(start_date_strin
 
     else:
         logger.info("Skipped the queuing of file upload tasks: max tasks already in progress.")
+
+
+# WACZ CONVERSION
+
+def format_size(bytes_val):
+    """
+    Converts bytes to KB and MB
+    """
+    kilobytes = bytes_val / 1024
+
+    if kilobytes < 1:
+        return f"{bytes_val} bytes"
+    elif kilobytes < 1024:
+        return f"{math.ceil(kilobytes)} KB"
+    else:
+        megabytes = kilobytes / 1024
+        return f"{megabytes:.2f} MB"
+
+
+def format_time(seconds_val):
+    """
+    Converts seconds to minutes
+    """
+    if seconds_val >= 60:
+        return f"{round(seconds_val / 60, 1)} minutes"
+    else:
+        return f"{math.ceil(seconds_val)} seconds"
+
+
+@shared_task
+def convert(input_path, output_folder, benchmark_log):
+    """
+    Converts WARC file to WACZ
+    Saves file in a temp folder
+    If successful, saves file in storage
+    """
+    start_time = time.time()
+    input_file_name = input_path.split('/')[-1]
+    output_path = f"{output_folder}/{input_file_name.split('.')[0]}.wacz"
+    js_wacz_call = subprocess.run(["npx", "js-wacz", "create", "-f", input_path, "-o", output_path],
+                                  capture_output=True, check=True, text=True)
+    end_time = time.time()
+    duration = format_time(end_time - start_time)
+    file_size = format_size(os.path.getsize(input_path))
+
+    with open(benchmark_log, 'a') as log_file:
+        row = {
+            "file_name": input_file_name,
+            "conversion_status": '',
+            "file_size": file_size,
+            "duration": duration,
```

Comment:
A suggestion: It might be worth adding durations and sizes to the CSV in a format that allows us to filter and sort easily. 

It will be for example difficult to sort / plot a column that contains strings such as "322 KB" and "1.2 MB"

Reply:
Added the raw fields as well as keeping the formatted fields.